### ISSUE TITLE
Linux: O3DE GitHub Development branch fails building

### ISSUE THREAD

**Issue by LB-ArturSiewniak** [o3de/o3de#6129]
**Describe the bug**
Attempting to build the Development branch on the Linux platform causes an XcbInputDeviceKeyboard errors to stop the building process. 

It should be noted that building the Stabilization_2110 (6480041) on Linux does not have this issue. 

It should also be noted that building on the Development branch on Windows does not have this issue as well.

Please refer to the **Building log** section for errors that appear in the Terminal on failure.

**Steps to reproduce**
1. Download the O3DE GitHub repository Development branch.
2. Configure/build the engine.

**Expected behavior**
O3DE Development branch builds successfully.

**Actual behavior**
O3DE Development branch fails to build.

**Building log**
```
gpc-610@GPC-610:~/o3de/AutomatedTesting$ cmake --build build/linux --config profile --target $O3DE_PROJECT_NAME.GameLauncher Editor -j 6
[34/1466] Building CXX object o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/profile/Unity/unity_21_cxx.cxx.o
FAILED: o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/profile/Unity/unity_21_cxx.cxx.o 
/usr/bin/clang++-12 -DAZ_BUILD_CONFIGURATION_TYPE=\"profile\" -DAZ_ENABLE_DEBUG_TOOLS -DAZ_ENABLE_TRACING -DAZ_PROFILE_BUILD=1 -DLINUX -DLINUX64 -DNDEBUG -DOPENSSL_ENABLED -DPAL_TRAIT_LINUX_WINDOW_MANAGER_XCB -D_HAS_EXCEPTIONS=0 -D_PROFILE -D__linux__ -DCMAKE_INTDIR=\"profile\" -I/home/gpc-610/o3de/Code/Framework/AzFramework/. -I/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Linux -I/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common -I/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb -I/home/gpc-610/o3de/Code/Framework/AzCore/. -I/home/gpc-610/o3de/Code/Framework/AzCore/Platform/Linux -I/home/gpc-610/o3de/Code/Framework/AzCore/Platform/Common -I/home/gpc-610/o3de/Code/Framework/GridMate/. -I/home/gpc-610/o3de/Code/Framework/GridMate/Platform/Linux -isystem /home/gpc-610/.o3de/3rdParty/packages/Lua-5.3.5-rev5-linux/Lua/include -isystem /home/gpc-610/.o3de/3rdParty/packages/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include -isystem /home/gpc-610/.o3de/3rdParty/packages/RapidXML-1.13-rev1-multiplatform/RapidXML/include -isystem /home/gpc-610/.o3de/3rdParty/packages/zlib-1.2.11-rev5-linux/zlib/include -isystem /home/gpc-610/.o3de/3rdParty/packages/zstd-1.35-multiplatform/zstd/lib -isystem /home/gpc-610/.o3de/3rdParty/packages/cityhash-1.1-multiplatform/cityhash/src -isystem /home/gpc-610/.o3de/3rdParty/packages/OpenSSL-1.1.1b-rev2-linux/OpenSSL/include -isystem /home/gpc-610/.o3de/3rdParty/packages/lz4-1.9.3-vcpkg-rev4-linux/lz4/lz4/include -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Werror -Wno-inconsistent-missing-override -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-parentheses -Wno-reorder -Wno-switch -Wno-undefined-var-template -fPIC -msse4.1  -O2 -g -std=c++17 -MD -MT o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/profile/Unity/unity_21_cxx.cxx.o -MF o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/profile/Unity/unity_21_cxx.cxx.o.d -o o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/profile/Unity/unity_21_cxx.cxx.o -c /home/gpc-610/o3de/AutomatedTesting/build/linux/o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/Unity/unity_21_cxx.cxx
In file included from /home/gpc-610/o3de/AutomatedTesting/build/linux/o3de/Code/Framework/AzFramework/CMakeFiles/AzFramework.dir/Unity/unity_21_cxx.cxx:15:
In file included from /home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.cpp:12:
/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.h:20:11: error: redefinition of 'XcbInputDeviceKeyboard'
    class XcbInputDeviceKeyboard
          ^
/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_Linux.cpp:10:10: note: '/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.h' included multiple times, additional include site here
#include <AzFramework/XcbInputDeviceKeyboard.h>
         ^
/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.cpp:12:10: note: '/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.h' included multiple times, additional include site here
#include <AzFramework/XcbInputDeviceKeyboard.h>
         ^
/home/gpc-610/o3de/Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceKeyboard.h:20:11: note: unguarded header; consider using #ifdef guards or #pragma once
    class XcbInputDeviceKeyboard
          ^
1 error generated.
[39/1466] Building CXX object o3de/Code/Framework/AzCore/CMakeFiles/AzCore.dir/profile/Unity/unity_0_cxx.cxx.o
ninja: build stopped: subcommand failed.
```

**Found in Branch**
Development (cfb426b)

**Desktop**
Device: PC
OS: Linux
Version: Ubuntu 20.04.3 LTS
CPU AMD Ryzen 5 3600
GPU NVIDIA GeForce RTX 2060 SUPER
Memory 16GB

**Comment by amzn-pratikpa** [comment]
@burelc-amzn already fixed this with https://github.com/o3de/o3de/pull/6091

**Comment by LB-ArturSiewniak** [comment]
Issue verified as fixed on Development (9698047) build.
